oggm/core/flowline.py

```diff
@@ -603,17 +620,47 @@ def run_until_and_store(self, y1, run_path=None, diag_path=None):
         nm = len(monthly_time)
         sects = [(np.zeros((ny, fl.nx)) * np.NaN) for fl in self.fls]
         widths = [(np.zeros((ny, fl.nx)) * np.NaN) for fl in self.fls]
-        diag_ds = xr.Dataset(coords=OrderedDict(time=('time', monthly_time),
-                                                year=('time', yrs),
-                                                month=('time', months),
-                                                ))
+        diag_ds = xr.Dataset()
+
+        # Global attributes
+        diag_ds.attrs['description'] = 'OGGM model output'
+        diag_ds.attrs['oggm_version'] = __version__
+        diag_ds.attrs['calendar'] = '365-day no leap'
+        diag_ds.attrs['creation_date'] = strftime("%Y-%m-%d %H:%M:%S",
+                                                  gmtime())
+
+        # Coordinates
+        diag_ds.coords['time'] = ('time', monthly_time)
+        diag_ds.coords['hydro_year'] = ('time', yrs)
+        diag_ds.coords['hydro_month'] = ('time', months)
+        diag_ds.coords['calendar_year'] = ('time', cyrs)
+        diag_ds.coords['calendar_month'] = ('time', cmonths)
+
+        diag_ds['time'].attrs['description'] = 'Floating hydrological year'
+        diag_ds['hydro_year'].attrs['description'] = 'Hydrological year'
+        diag_ds['hydro_month'].attrs['description'] = 'Hydrological month'
+        diag_ds['calendar_year'].attrs['description'] = 'Calendar year'
+        diag_ds['calendar_month'].attrs['description'] = 'Calendar month'
+
+        # Variables and attributes
         diag_ds['volume_m3'] = ('time', np.zeros(nm) * np.NaN)
+        diag_ds['volume_m3'].attrs['description'] = 'Total glacier volume'
+        diag_ds['volume_m3'].attrs['unit'] = 'm 3'
         diag_ds['area_m2'] = ('time', np.zeros(nm) * np.NaN)
+        diag_ds['area_m2'].attrs['description'] = 'Total glacier area'
+        diag_ds['area_m2'].attrs['unit'] = 'm 2'
         diag_ds['length_m'] = ('time', np.zeros(nm) * np.NaN)
+        diag_ds['length_m'].attrs['description'] = 'Glacier length'
+        diag_ds['length_m'].attrs['unit'] = 'm 3'
         diag_ds['ela_m'] = ('time', np.zeros(nm) * np.NaN)
+        diag_ds['ela_m'].attrs['description'] = ('Annual Equilibrium Line '
+                                                 'Altitude')
```

Comment:
I think that the Equilibrium Line Altitude is by definition annual, therefore I believe that the word 'annual' is redundant in this case.

Reply:
Correct. I'll keep it that way though: because of implementation details, it is still stored at a monthly time step, so to make it clear I added the "Annual"

Reply:
I didn't think of that, but in that case it makes indeed a lot off sense to leave it like it is :)